src/development/vortex/development/utils/registry.py

```diff
@@ -0,0 +1,163 @@
+from typing import Dict, Any, Union
+from copy import copy
+import inspect
+
+class Registry:
+    def __init__(self, name: str, base_class=None):
+        self.base_class = base_class
+        self.name = name
+        self.modules: Dict[str,Any] = {}
+    
+    def get(self, key):
+        return self.modules[key]
+    
+    def __len__(self):
+        return len(self.modules)
+    
+    def __contains__(self, key: str):
+        return key in self.modules
+    
+    def __repr__(self):
+        cls_name = self.__class__.__name__
+        fmt = f"{cls_name}<{self.name}>"
+        return fmt
+    
+    def __getitem__(self, key: str):
+        """Added to behaves like dict
+
+        Args:
+            key (str): key
+
+        Returns:
+            type: type associated with "key"
+        """
+        return self.get(key)
+    
+    def register_module(self, module, name: Union[str,None]=None, force: bool=False, overwrite: bool=False):
+        # typecheck
+        if not isinstance(force,bool):
+            raise TypeError("expect force to be string type, got {}".format(type(force)))
+        # typecheck
+        if not isinstance(overwrite,bool):
+            raise TypeError("expect overwrite to be string type, got {}".format(type(overwrite)))
+        # typecheck
+        # if not (inspect.isfunction(module) or inspect.isclass(module)):
+        if not inspect.isclass(module):
```

Comment:
why is here not also accepting function? it could also accept function right?

Reply:
well yes, but somehow I think it is better to limit to class for clarity and simplicity.

Limiting to only class, `create_bla2` semantically return an object, while allowing function is rather wild. Any thoughts why we need to support functions?

Reply:
I think there are some cases for creating model in family better via function, like those when creating backbone. as far as I know, detr also have a derived model as well, so the ability to support function is really useful in my opinion

Reply:
ok please wait